server/src/lib/model/user-client.ts

```diff
@@ -245,17 +252,56 @@ const UserClient = {
     return true;
   },
 
-  // update the challenge_team of the user who is already signed up but have not joined any team
-  // [BUG]: there are exceptions if the team doesn't exist
+  // enroll an unenrolled but registered user
+  // [BUG]: there are exceptions if the team or challenge doesn't exist
   // TODO(riley): Hook this up to a constants file.
-  async updateChallengeTeam(
+  async enrollRegisteredUser(
     email: string,
-    challenge_team: string
+    challenge: string,
+    team: string,
+    invite: string
   ): Promise<boolean> {
-    return await db.query(
-      'UPDATE user_clients SET challenge_team = ? WHERE email = ? AND has_login AND challenge_team IS NULL',
-      [challenge_team, email]
-    );
+    if (email && challenge && team) {
+      let client_id = await Promise.all([UserClient.findClientId(email)]);
+      if (client_id) {
+        // Generate a random, anonymized, and unique invite ID for enrollments.
+        // HACK(riley/can): Can we move this entirely into the DB?
```

Comment:
mysql has a function named uuid(), just what we need. Haven't successfully run this new code yet.

Reply:
Oh wow 😂 amazing!!! `UUID()` is a bit long for an invite code, but maybe we can use `UUID_SHORT()` and convert it from decimal => base 36 on the server?

Practically, `UUID_SHORT()` is close to just using `AUTO INCREMENT`, but I feel fine about that. Thanks for calling my attention to all this!!